### ISSUE TITLE
Starting service during autopkgtest fails

### ISSUE THREAD

**Issue by geor-g** [lamby/travis.debian.net#45]
I'm searching for input regarding the following problem:

- One of the [tests](https://0xacab.org/schleuder/schleuder/blob/debian/unstable/debian/tests/check-status) of schleuder checks if the API daemon is running.
- The provided init script doesn't work, because of `policy-rc.d` I guess. However, also starting the service manually (like in the linked file, above) [doesn't work](https://travis-ci.org/schleuder/schleuder3/builds/243348991#L2329).

Any idea?

**Comment by lamby** [comment]
Is this a ``debian/rules`` test or an ``autopkgtest`` ?

We could (should?) "revert" the change to policy-rc.d for autopkgtest - seems wrong they are disabled there.

Alternatively, a global option that reverted policy-rc.d for the building phase (after installing build-deps with it as it currently is!) could work.

What do you think?

**Comment by geor-g** [comment]
> Is this a `debian/rules` test or an `autopkgtest`?

The later.

> We could (should?) "revert" the change to policy-rc.d for autopkgtest - seems wrong they are disabled there.
>
> Alternatively, a global option that reverted policy-rc.d for the building phase (after installing build-deps with it as it currently is!) could work.
>
> What do you think?

I think such an option (controlled by an env var, like the other options?) would be valuable.

However, I'm still unsure what my specific problem is about: Even starting the daemon manually via `service ...` doesn't work; in contrast, `exim` gets started, just via the `init.d` script, as the output of `netstat` in the second link above shows. Any idea regarding this?

**Comment by lamby** [comment]
> I think such an option (controlled by an env var, like the other options?) would be valuable.

I *was* going to go ahead and implement disabling of ``policy-rc.`` but without an environment variable to start with — don't want to add too many... but it's already implemented here: f9e8148c909a6c2f40db714f58174f304b84cf12

> Any idea regarding this?

(No, sorry. You'd have to play with Travis itself alas to work out why this is happening.)

**Comment by geor-g** [comment]
Alright, thanks for your time.

Because I've spent quite some time already to figure out that the problem is about, and not finding the cause, I've just thought about disabling this specific test on Travis, if a specific `env var` exists, for example. Travis defines quite some [of these](https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables), however, inside the docker container, these aren't set. I got that you don't like to add more options, but could we maybe set just one, `TRAVIS=true` for example?
